drivers/hmis/app/graphql/mutations/publish_form_definition.rb

```diff
@@ -23,7 +23,9 @@ def resolve(id:)
       )
 
       definition.status = Hmis::Form::Definition::PUBLISHED
-
+      # Ensure HUD requirements are set correctly (if applicable). This could mutate the definition.
+      definition.set_hud_requirements
```

Comment:
We have no validation to ensure that the content of the "hud fragments" are not modified inappropriately. We rely on the frontend to make sure that doesn't happen.

Looking at this with fresh eyes, it feels kind of arbitrary that we decided to strictly enforce/check that `rule` doesn't change, but we don't bother to check if the actual structure of the group (or item) matches what is expected. (See for example my test where I make "c3 education status" be a boolean).